src/parser.rs

```diff
@@ -1668,7 +1668,13 @@ impl Parser {
         if self.parse_keyword("COLUMNS") {
             self.parse_show_columns()
         } else {
-            self.expected("COLUMNS", self.peek_token())
+            self.parse_show_objects().ok_or(
```

Comment:
Below I advocate for removing this block in general, but if it were to stay this should be `ok_or_else(|| ` so that we don't construct the error unless we actually hit the else branch.